src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/DefaultHttpClient.java

```diff
@@ -19,14 +19,14 @@
 final class DefaultHttpClient extends HttpClient {
     private OkHttpClient client = null;
 
-    public DefaultHttpClient() {
-        this(0, null);
+    public DefaultHttpClient(Action1<OkHttpClient.Builder> configureBuilder) {
+        this(0, null, configureBuilder);
     }
 
     public DefaultHttpClient cloneWithTimeOut(int timeoutInMilliseconds) {
         OkHttpClient newClient = client.newBuilder().readTimeout(timeoutInMilliseconds, TimeUnit.MILLISECONDS)
                 .build();
-        return new DefaultHttpClient(timeoutInMilliseconds, newClient);
+        return new DefaultHttpClient(timeoutInMilliseconds, newClient, null);
```

Comment:
Should we store the configureBuilder and reapply it for cloned clients? Long term, we should add timeouts with something other than cloneWithTimeOut, but for now, it seems broken not to.

Reply:
The cloned client gets the same configuration applied. It's using the same builder internally effectively.

Reply:
Wait a minute... It looks like `timeoutInMilliseconds` is [completely ignored](https://github.com/dotnet/aspnetcore/blob/b5690db723351504fe96785b0bbebd0b0699f757/src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/DefaultHttpClient.java#L41) when a non-null client is passed in. I guess there's no need to pass in `configureBuilder` if it wouldn't be called anyway, but this looks completely broken to me. How do timeouts work if timeoutInMilliseconds is just being thrown away? Did it ever work? Am I missing something obvious?

Reply:
The constructors are a bit odd here, could use some cleanup. The timeout is actually used 2 lines above and unused when passed in here.

Reply:
We're going to reapply the configureBuilder then, right?

Reply:
`newBuilder` already copies the settings that were applied with the `configureBuilder`.